src/proxy/go/go.c

```diff
@@ -255,6 +255,13 @@ int proxy_go_input_cleanup(struct flb_plugin_proxy *ctx,
     if (plugin->cb_cleanup) {
         ret = plugin->cb_cleanup(allocated_data);
     }
+    else {
+        /* If cleanup callback is not registered, we need to cleanup
+         * allocated memory on fluent-bit side. */
```

Comment:
Is this attempting to free a NULL pointer?

Reply:
No. Without cleanup callback, there is no freeing resource mechanism. These lines tend to free allocated resources where are allocated on input callback.

Reply:
I changed to use not NULL in https://github.com/fluent/fluent-bit/pull/6823/commits/c7c074398fb875e903cc111ecac236e971036299.